File: rest.py
import requests
from requests.auth import HTTPBasicAuth

def make_rest_call(username, password):
    url = "http://localh2-env.eba-rpvpzjqz.eu-central-1.elasticbeanstalk.com/plug/48551917CE6C"

    # ophalen en verwerken van de username en password
    auth = HTTPBasicAuth(username, password)

    try:
        #call maken
        response = requests.get(url, auth=auth)

        # als er een connectie is print de response uit anders error printen
        if response.status_code == 200:
            print("REST call successful.")
            print("Response:")
            print(response.text)
        else:
            print(f"Failed to make the REST call. Status Code: {response.status_code}")
            print("Response:")
            print(response.text)

    except Exception as e:
        print(f"An error occurred: {e}")

if __name__ == "__main__":
    # username en password via input zoat niemand erbij kan
    username = input("Enter username: ")
    password = input("Enter password: ")

    make_rest_call(username, password)

File: rest.php
<?php

function makeRestCall($username, $password) {
    $url = "http://localh2-env.eba-rpvpzjqz.eu-central-1.elasticbeanstalk.com/plug/48551917CE6C";

    // ophalen en verwerken van de username en password
    $auth = base64_encode("$username:$password");
    $headers = array("Authorization: Basic $auth");

    try {
        // call maken
        $ch = curl_init($url);
        curl_setopt($ch, CURLOPT_HTTPHEADER, $headers);
        curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);

        $response = curl_exec($ch);
        $httpCode = curl_getinfo($ch, CURLINFO_HTTP_CODE);

        // als er een connectie is print de response uit anders error printen
        if ($httpCode == 200) {
            echo "REST call successful." . PHP_EOL;
            echo "Response:" . PHP_EOL;
            echo $response;
        } else {
            echo "Failed to make the REST call. Status Code: $httpCode" . PHP_EOL;
            echo "Response:" . PHP_EOL;
            echo $response;
        }

        curl_close($ch);

    } catch (Exception $e) {
        echo "An error occurred: " . $e->getMessage() . PHP_EOL;
    }
}

// username en password via input zoat niemand erbij kan
$username = readline("Enter username: ");
$password = readline("Enter password: ");

makeRestCall($username, $password);

?>
